src/AbstractSerializer.php

```diff
@@ -114,15 +114,8 @@ protected static function splitStream(StreamInterface $stream)
             $headers[$currentHeader][] = $value . ltrim($line);
         }
 
-        $body = new Stream('php://temp', 'wb+');
-        if (! $stream->eof()) {
-            while ($data = $stream->read(4096)) {
-                $body->write($data);
-            }
-            $body->rewind();
-        }
-
-        return [$headers, $body];
+        // use RelativeStream to avoid copying initial stream into memory
+        return [$headers, new RelativeStream($stream, $stream->tell())];
```

Comment:
@mtymek that's a separate issue. What you are actually verifying here is that you're not sequentially scanning through a huge file output, but are instead lazily evaluating it.

Asserting that a `RelativeStream` is returned gives you no such guarantee.